Kernel/Tasks/Thread.cpp

```diff
@@ -1401,6 +1401,10 @@ bool Thread::track_lock_acquire(LockRank rank)
     // Nothing to do for locks without a rank.
     if (rank == LockRank::None)
         return false;
+    if (rank == LockRank::Mutex && m_lock_rank_mask != LockRank::None) {
```

Comment:
This check could be generalized if we had a notion in the bitfield about which ranks are for mutexes and which are for spinlocks :^) (not necessary for now)